### PR TITLE
Completes F5.1, F5.3, F5.4, F5.10 for frontend

### PR DESCRIPTION
**Pull request by github-amanda**
1. Added reset page and reset password functionality. 
image
image

2. Modified layout of room page (for full implementation, refer to pr#57).
imageimage
image

3. Added a "Leave Queue" functionality to the matching page, which will route the user back to landing page and disconnect the socket upon activated. 
image
 
Completes F5.1, F5.3, F5.4 and F5.10 of the sprint plan.

### CODE REVIEW

File: frontend/src/components/NavigationBar.js [hunk 1]
```diff
@@ -76,20 +78,38 @@ function NavigationBar({ isAuthenticated }) {
...
  
-    const handleChangePasswordOnClick = () => {
+    const handleChangePasswordOnClick = async(event) => {
         setAnchorEl(null);
-        // Triggers change password! 
-    } 
+        event.preventDefault();
+        const refresh_token = cookies["refresh_token"]
+        const username = jwtDecode(refresh_token).username;
+        const res = await axios.put(URL_USER_SVC_UPDATEPASSWORD, { username, newPassword })
```
Comment (ngzhenteng) [inline frontend/src/components/NavigationBar.js:86]: Could i check if we should be using axiosApiInstance to make these requests? Or you already are using it? Thanks

Comment (github-amanda) [reply frontend/src/components/NavigationBar.js:86]: I'm not using it here

**Comment by ngchisern** [comment]
Just a suggestion, is it better if the question and code editor window are next to each other?

**Review by yongxiangng — CHANGES_REQUESTED** [review]
let's fix the ci (it's failing because of some unused variables) before pushing it in 🙇

**Comment by github-amanda** [comment]
> Just a suggestion, is it better if the question and code editor window are next to each other?

I think both are fine, I'm following the layout used in the demo shown to us.

**Review by yongxiangng — APPROVED** [review]

**Review by yongxiangng — APPROVED** [review]
LGTM! thanks

**Review by ngzhenteng — APPROVED** [review]
LGTM thanks you!

**Pull request merged**